manifests/machineconfig.crd.yaml

```diff
@@ -39,205 +41,401 @@ spec:
     # shortNames allow shorter string to match your resource on the CLI
     shortNames:
     - mc
-  validation:
-    openAPIV3Schema:
+  "validation":
+    "openAPIV3Schema":
+      description: MachineConfig defines the configuration for a machine
       type: object
       properties:
         apiVersion:
-          description: "APIVersion defines the versioned schema of this representation
+          description: 'APIVersion defines the versioned schema of this representation
             of an object. Servers should convert recognized schemas to the latest
-            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources"
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
           type: string
         kind:
-          description: "Kind is a string value representing the REST resource this
+          description: 'Kind is a string value representing the REST resource this
             object represents. Servers may infer this from the endpoint the client
-            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds"
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
           type: string
         metadata:
           type: object
         spec:
-          description: "spec hold the intent of how this operator should behave"
+          description: MachineConfigSpec is the spec for MachineConfig
           type: object
           properties:
             config:
-              description: "config contains options related to the configuration"
+              description: Config is a Ignition Config object.
               type: object
+              required:
+              - ignition
               properties:
                 ignition:
-                  description: "ignition section contains metadata about the configuration itself"
                   type: object
                   properties:
+                    config:
```

Comment:
just a note: We won't get any of the CRD schema validation for RawExtension :(
Spec2 and Spec3 are different structs, so we'd either have to choose one for (manually added) validation or none and do validation in the Installer/MCO.

Reply:
It's probably ok since we will have error reporting via the MCP regarding if a conf is invalid anyways. This generation hopefully won't be needed here and can go directly to the API repo once the conversion is in.

For the time being https://github.com/openshift/machine-config-operator/pull/1485 is good to unblock things.